target_postgres/db_sync.py

```diff
@@ -119,6 +119,9 @@ def __init__(self, connection_config, stream_schema_message):
         self.rejected_count = 0
 
     def open_connection(self):
+        if self.connection_config['url']:
+            return psycopg2.connect(self.connection_config['url'])
```

Comment:
To remain compatibility with existing `config.json` files, we can't leave the `KeyError` that would be raised if the `'url'` key isn't present unhandled here. How about:

```py
try:
    url = self.connection_config['url']
    return psycopg2.connect(url)
except KeyError:
    pass
```

Reply:
🤦 my poor python skills are showing

Reply:
@rabidaudio Interestingly enough, my `try`/`except` suggestion is actually more Pythonic than yours: https://stackoverflow.com/a/11360880 ;)

Either way, both of our implementations forget that if `self.connection_config['url']` is set but `None` (~ `null`), we should not actually try to use it! So we should add an `if url` check.

Reply:
Thanks for the learning opportunity :)

Should I squash these commits?

Reply:
@rabidaudio I can "Squash and Merge" once it's ready, so no need :)